### PR TITLE
Refs #30947 -- Changed tuples to lists where appropriate.

### PR DESCRIPTION
**Pull request by mgax**
The patch is split into 4 commits, hopefully making it easier to review.

Most of the tuples that got changed were found by searching the files with a regexp. Only the code in `django/contrib/*/models*` was searched because that's the scope defined in the ticket. TBH I'm not sure if the changes to the docs should have been part of the ticket, but here they are :)

### CODE REVIEW

File: docs/ref/contrib/admin/index.txt [hunk 1]
```diff
@@ -766,7 +766,7 @@ subclass::
                   return self.first_name + ' ' + self.last_name
 
           class PersonAdmin(admin.ModelAdmin):
-              list_display = ('full_name',)
+              list_display = ['full_name',]
```
Comment (felixxm) [inline docs/ref/contrib/admin/index.txt:769]:
```suggestion
              list_display = ['full_name']
```

File: docs/ref/contrib/admin/index.txt [hunk 2]
```diff
@@ -1736,12 +1736,12 @@ templates used by the :class:`ModelAdmin` views:
...
                 if db_field.name == "status":
-                    kwargs['choices'] = (
+                    kwargs['choices'] = [
                         ('accepted', 'Accepted'),
                         ('denied', 'Denied'),
-                    )
+                    ]
                     if request.user.is_superuser:
-                        kwargs['choices'] += (('ready', 'Ready for deployment'),)
+                        kwargs['choices'] += [('ready', 'Ready for deployment')]
```
Comment (felixxm) [inline docs/ref/contrib/admin/index.txt:1744]:
```suggestion
                        kwargs['choices'].append(('ready', 'Ready for deployment'))
```

**Comment by mgax** [comment]
@felixxm thanks!

File: docs/ref/forms/fields.txt [hunk 3]
```diff
@@ -1158,7 +1158,7 @@ Slightly complex built-in ``Field`` classes
                             validators=[RegexValidator(r'^[0-9]+$', 'Enter a valid extension.')],
                             required=False,
                         ),
-                    )
+                    ]
```
Comment (felixxm) [inline docs/ref/forms/fields.txt:1161]: This is document as a tuple and should remain a tuple. Reverted.

**Review by felixxm — COMMENTED** [review]
@mgax Thanks :+1:

**Comment by akx** [comment]
This will break (broke for me in CI) uses such as

```python
class CustomUserAdmin(DjangoUserAdmin):
   inlines = DjangoUserAdmin.inlines + ['field', ...]
```
since
> TypeError: can only concatenate tuple (not "list") to tuple

Easily fixed, of course, by changing that `list` to a tuple (and perhaps it shouldn't have been a list in my code in the first place), or by splatting (which doesn't care about the iterable type), e.g.

```python
class CustomUserAdmin(DjangoUserAdmin):
   inlines = (*DjangoUserAdmin.inlines, 'field')
```

but still... an unwelcome surprise for a 4.1.0 -> 4.1.1 release :)

**Comment by felixxm** [comment]
> This will break (broke for me in CI) uses such as

How this could break your CI? :thinking: First of all it's almost a purely documentation change. Secondly, it was not backported to Django 4.1.

**Comment by akx** [comment]
> How this could break your CI? 🤔 [...] Secondly, it was not backported to Django 4.1.

Oops, sorry, I jumped the gun there looking at `blame` – it was actually https://github.com/django/django/pull/15270 that broke things. I couldn't use 4.1.0 at all due to #15906 :)

**Comment by felixxm** [comment]
> Oops, sorry, I jumped the gun there looking at `blame` – it was actually #15270 that broke things. I couldn't use 4.1.0 at all due to #15906 :)

It's documented as a small breaking change, see [release notes](https://docs.djangoproject.com/en/4.1/releases/4.1/#miscellaneous):
> "_The [ModelAdmin.actions](https://docs.djangoproject.com/en/4.1/ref/contrib/admin/#django.contrib.admin.ModelAdmin.actions) and [inlines](https://docs.djangoproject.com/en/4.1/ref/contrib/admin/#django.contrib.admin.ModelAdmin.inlines) attributes now default to an empty tuple rather than an empty list to discourage unintended mutation."_

**Comment by akx** [comment]
@felixxm Yep, thanks :) I swear I did read it there 😁 but as said, I had to skip 4.1.0 altogether due to the accidental GDAL dependency so this never cropped up then. All good now, a bit of refactoring and I have `*` splats everywhere :)

**Pull request merged**